clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -584,7 +599,7 @@ private void validate(String name, Map<String, Object> parsed, Map<String, Confi
      * @param type  The expected type
      * @return The parsed object
      */
-    private Object parseType(String name, Object value, Type type) {
+    public static Object parseType(String name, Object value, Type type) {
```

Comment:
The usages in `ConnectorConfig` are around parsing a `Type.LIST` or `Type.CLASS`, to reuse the existing parsing logic for those...